### PR TITLE
dockerfile

### PR DESCRIPTION
**Pull request by tkv29**
sudo docker run -p 8000:8000

### CODE REVIEW

File: README.md
Comment (PitButtchereit) [inline README.md:1]: Can we also get this badge for our repo?

grafik

Comment (PitButtchereit) [reply README.md:1]: Let's state somewhere that we are using GPT 3.5 turbo so people know how much they will get charged. Maybe under the requirements section.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
-More information about the project will be released soon.
+## Requirements
+To run TracEX successfully, it is essential to obtain an OpenAI API key with adequate credits. TracEX integrates the OpenAI API to leverage Large Language Models (LLMs) for extracting relevant information from unstructured text. Without a valid API key and sufficient balance, the extraction process cannot be performed.
 
+## Installation using Docker
+**Option 1: Using a Pre-built Docker Image** \
+The easiest way to run a local instance of TracEX is using the provided Docker image.
 
-## Set Up Guide
+1. Install Docker: Ensure that you have Docker installed on your system. If you haven't installed it yet, please follow the official Docker installation guide for your operating system: Docker Installation.
```
Comment (PitButtchereit) [inline README.md:26]: Let's provide a link to the docker installation guide here.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
+1. Download the Latest Docker Image: Download the latest TracEX Docker image from the provided link: docker image
```
Comment (PitButtchereit) [inline README.md:27]: I think we are missing the link to the docker image here.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,48 @@
...
+1. Load the Docker Image: Open a terminal or command prompt and navigate to the directory where you downloaded the Docker image file. Run the following command to load the image: `docker load -i tracex.tar`\
+Note: Depending on your system configuration, you may need to run this command with sudo privileges.
+1. Run the Docker Container: After the image is successfully loaded, run the following command to start the TracEX container: `docker run -p 8000:8000 tracex`\
+This command will start the container and map port 8000 from the container to port 8000 on your local machine. Again, you may need to use sudo depending on your system setup.
```
Comment (PitButtchereit) [inline README.md:31]:
```suggestion
This command will start the container and map port 8000 from the container to port 8000 on your local machine. Again, you may need to use `sudo` depending on your system setup.
```

File: README.md [hunk 4]
```diff
@@ -1,16 +1,48 @@
...
+1. Access TracEX: Open a web browser and navigate to http://localhost:8000/. This will bring you to the TracEX application, where you can enter your OpenAI Key and start extracting event logs.
+
+**Option 2: Building the Docker Image from Source** \
+Alternatively, you can build the Docker image from the TracEX source code.
+
+1. Clone the TracEX Repository: Open a terminal or command prompt and navigate to the directory where you want to clone the TracEX repository. Run the following command to clone the repository: git clone https://github.com/bptlab/TracEX
```
Comment (PitButtchereit) [inline README.md:37]:
```suggestion
1. Clone the TracEX Repository: Open a terminal or command prompt and navigate to the directory where you want to clone the TracEX repository. Run the following command to clone the repository: `git clone https://github.com/bptlab/TracEX`
```

File: README.md [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
+1. Navigate to the TracEX Directory: Change your current directory to the cloned TracEX repository: `cd TracEX`
+1. Build the Docker Image: Run the following command to build the TracEX Docker image: `docker build -t tracex .`\
+Note: Depending on your system configuration, you may need to run this command with sudo privileges.
```
Comment (PitButtchereit) [inline README.md:40]:
```suggestion
Note: Depending on your system configuration, you may need to run this command with `sudo` privileges.
```

File: README.md [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+1. Run the Docker Container: After the image is successfully built, run the following command to start the TracEX container: `docker run -p 8000:8000 tracex`\
+This command will start the container and map port 8000 from the container to port 8000 on your local machine. Again, you may need to use sudo depending on your system setup.
```
Comment (PitButtchereit) [inline README.md:42]:
```suggestion
This command will start the container and map port 8000 from the container to port 8000 on your local machine. Again, you may need to use `sudo` depending on your system setup.
```

File: README.md
Comment (PitButtchereit) [inline README.md:1]: Under ###Download write "Use git and run `git clone https://github.com/bptlab/TracEX` in the desired directory _(Using e.g. Git Bash)_"

File: README.md [hunk 7]
```diff
@@ -19,16 +51,31 @@ More information about the project will be released soon.
 ### Installation
 - navigate to the root directory of TracEX in your terminal
 - run `install-dependencies-unix.sh` or `install-dependencies-windows.ps1`, based on your operating system _(Using e.g. Terminal)_
-- run `python tracex/manage.py migrate` to update the database and apply all changes stored in the `migrations` folder
+- run `python tracex_project/manage.py migrate` to update the database and apply all changes stored in the `migrations` folder
```
Comment (PitButtchereit) [inline README.md:54]:
```suggestion
- run `python tracex_project/manage.py migrate` to update the database and apply all changes stored in the `migrations/` folder
```

**Review by PitButtchereit — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -1,16 +1,48 @@
...
+## Key Features
+- **Extraction Pipeline**: A robust pipeline to clean, process, and extract data from natural language text.
+- **Patient Journey Generator**: Generates comprehensive patient journeys based on randomized cohort data.
+- **Database**: Stores patient journeys and related extraction results for easy access and analysis.
+- **Metrics and Evaluation Tool**: Evaluates the accuracy and effectiveness of the extraction process and allows for analysis of exctraction results.
+- **Intuitive UI**: User-friendly interface for you to interact with the tool and visualize results.
 
-More information about the project will be released soon.
+## Requirements
+To run TracEX successfully, it is essential to obtain an OpenAI API key with adequate credits. TracEX integrates the OpenAI API to leverage Large Language Models (LLMs) for extracting relevant information from unstructured text. Without a valid API key and sufficient balance, the extraction process cannot be performed.
```
Comment (FR-SON) [inline README.md:20]:
```suggestion
To run TracEX successfully, it is essential to obtain an OpenAI API key with adequate credits. TracEX integrates the OpenAI API to leverage Large Language Models (LLMs) for extracting relevant information from unstructured text. Without a valid API key and sufficient balance, the extraction process cannot be performed. The current prices for API can be looked up at [OpenAI Pricing](https://openai.com/api/pricing/).
```

**Review by PitButtchereit — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -1,34 +1,83 @@
...
+## Key Features
+- **Extraction Pipeline**: A robust pipeline to clean, process, and extract data from natural language text.
+- **Patient Journey Generator**: Generates comprehensive patient journeys based on randomized cohort data.
+- **Database**: Stores patient journeys and related extraction results for easy access and analysis.
+- **Metrics and Evaluation Tool**: Evaluates the accuracy and effectiveness of the extraction process and allows for analysis of exctraction results.
+- **Intuitive UI**: User-friendly interface for you to interact with the tool and visualize results.
 
-More information about the project will be released soon.
+## Requirements
+To run TracEX successfully, it is essential to obtain an OpenAI API GPT 3.5 Turbo key with adequate credits. TracEX integrates the OpenAI API to leverage Large Language Models (LLMs) for extracting relevant information from unstructured text. Without a valid API key and sufficient balance, the extraction process cannot be performed. The current prices for API can be looked up at [OpenAI Pricing](https://openai.com/api/pricing/).
```
Comment (FR-SON) [inline README.md:21]:
```suggestion
To run TracEX successfully, it is essential to obtain an OpenAI API key with adequate credits. TracEX integrates the OpenAI API to leverage Large Language Models (LLMs) for extracting relevant information from unstructured text. Without a valid API key and sufficient balance, the extraction process cannot be performed. The current prices for API can be looked up at [OpenAI Pricing](https://openai.com/api/pricing/).
```

File: README.md [hunk 10]
```diff
@@ -1,34 +1,83 @@
...
 ### Download
 
-- Use git and run "git clone https://github.com/bptlab/TracEX" in the desired directory _(Using e.g. Git Bash)_
+- Use git and run `git clone https://github.com/bptlab/TracEX` in the desired directory _(Using e.g. Git Bash)_
 
 ### Installation
 - navigate to the root directory of TracEX in your terminal
 - run `install-dependencies-unix.sh` or `install-dependencies-windows.ps1`, based on your operating system _(Using e.g. Terminal)_
-- run `python tracex/manage.py migrate` to update the database and apply all changes stored in the `migrations` folder
+- run `python tracex_project/manage.py migrate` to update the database and apply all changes stored in the `migrations/` folder
```
Comment (FR-SON) [inline README.md:55]:
```suggestion
    - If you are using macOS, please use [homebrew](https://brew.sh/) or any other package manager to install the listed dependencies manually.
- run `python tracex_project/manage.py migrate` to update the database and apply all changes stored in the `migrations/` folder
```

**Review by FR-SON — APPROVED** [review]

**Pull request merged**